Convert module files on WCOSS2 and Hera to Lua format

XianwuXue-NOAA opened the pull request:
Fixes  #59

XianwuXue-NOAA committed:
Rename module files for WCOSS2 to *.lua

 On branch feature/ops_wcoss2_lua
 Changes to be committed:
	renamed:    modulefiles/gefs/gefs_anom2_fcst.wcoss2 -> modulefiles/gefs/gefs_anom2_fcst.wcoss2.lua
	renamed:    modulefiles/gefs/gefs_ensstat.wcoss2 -> modulefiles/gefs/gefs_ensstat.wcoss2.lua
	renamed:    modulefiles/gefs/gefs_nemsio2nc.wcoss2 -> modulefiles/gefs/gefs_nemsio2nc.wcoss2.lua
	renamed:    modulefiles/gefs/gefs_nstgen.wcoss2 -> modulefiles/gefs/gefs_nstgen.wcoss2.lua
	renamed:    modulefiles/gefs/global_ensadd.wcoss2 -> modulefiles/gefs/global_ensadd.wcoss2.lua
	renamed:    modulefiles/gefs/global_ensppf.wcoss2 -> modulefiles/gefs/global_ensppf.wcoss2.lua
	renamed:    modulefiles/gefs/wave_stat.wcoss2 -> modulefiles/gefs/wave_stat.wcoss2.lua

Refs: #59

XianwuXue-NOAA committed:
Convert module files for WCOSS2 to Lua

 On branch feature/ops_wcoss2_lua
 Changes to be committed:
	modified:   modulefiles/gefs/gefs_anom2_fcst.wcoss2.lua
	modified:   modulefiles/gefs/gefs_ensstat.wcoss2.lua
	modified:   modulefiles/gefs/gefs_nemsio2nc.wcoss2.lua
	modified:   modulefiles/gefs/gefs_nstgen.wcoss2.lua
	modified:   modulefiles/gefs/global_ensadd.wcoss2.lua
	modified:   modulefiles/gefs/global_ensppf.wcoss2.lua
	modified:   modulefiles/gefs/wave_stat.wcoss2.lua
	modified:   sorc/build_gefs_anom2_fcst.sh
	modified:   sorc/build_gefs_ensstat.sh
	modified:   sorc/build_gefs_nemsio2nc.sh
	modified:   sorc/build_gefs_nstgen.sh
	modified:   sorc/build_global_ensadd.sh
	modified:   sorc/build_global_ensppf.sh
	modified:   sorc/build_wave_stat.sh

Refs: #59

XianwuXue-NOAA committed:
Rename module files to lua for wcoss_dell_p3

 On branch feature/ops_wcoss2_lua
 Changes to be committed:
	renamed:    modulefiles/gefs/gefs_anom2_fcst.wcoss_dell_p3 -> modulefiles/gefs/gefs_anom2_fcst.wcoss_dell_p3.lua
	renamed:    modulefiles/gefs/gefs_ensstat.wcoss_dell_p3 -> modulefiles/gefs/gefs_ensstat.wcoss_dell_p3.lua
	renamed:    modulefiles/gefs/gefs_nemsio2nc.wcoss_dell_p3 -> modulefiles/gefs/gefs_nemsio2nc.wcoss_dell_p3.lua
	renamed:    modulefiles/gefs/gefs_nstgen.wcoss_dell_p3 -> modulefiles/gefs/gefs_nstgen.wcoss_dell_p3.lua
	renamed:    modulefiles/gefs/global_ensadd.wcoss_dell_p3 -> modulefiles/gefs/global_ensadd.wcoss_dell_p3.lua
	renamed:    modulefiles/gefs/global_ensppf.wcoss_dell_p3 -> modulefiles/gefs/global_ensppf.wcoss_dell_p3.lua
	renamed:    modulefiles/gefs/wave_stat.wcoss_dell_p3 -> modulefiles/gefs/wave_stat.wcoss_dell_p3.lua
	renamed:    modulefiles/gefs/gefs_wcoss_dell_p3.ver -> rocoto/dev/versions/build_wcoss_dell_p3.ver

Refs: #59

XianwuXue-NOAA committed:
Convert module files for wcoss_dell_p3 to lua

 On branch feature/ops_wcoss2_lua
 Changes to be committed:
	modified:   modulefiles/gefs/gefs_anom2_fcst.wcoss_dell_p3.lua
	modified:   modulefiles/gefs/gefs_ensstat.wcoss_dell_p3.lua
	modified:   modulefiles/gefs/gefs_nemsio2nc.wcoss_dell_p3.lua
	modified:   modulefiles/gefs/gefs_nstgen.wcoss_dell_p3.lua
	modified:   modulefiles/gefs/global_ensadd.wcoss_dell_p3.lua
	modified:   modulefiles/gefs/global_ensppf.wcoss_dell_p3.lua
	modified:   modulefiles/gefs/wave_stat.wcoss_dell_p3.lua
	modified:   sorc/build_gefs_anom2_fcst.sh
	modified:   sorc/build_gefs_ensstat.sh
	modified:   sorc/build_gefs_nemsio2nc.sh
	modified:   sorc/build_gefs_nstgen.sh
	modified:   sorc/build_global_ensadd.sh
	modified:   sorc/build_global_ensppf.sh
	modified:   sorc/build_wave_stat.sh

Refs: #59

XianwuXue-NOAA committed:
Merge branch 'feature/gefs.v12.3.2' into feature/ops_wcoss2_lua

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/ops_wcoss2_lua

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/dev_wcoss2_lua

XianwuXue-NOAA committed:
Remove dell_p3 setting

 On branch feature/dev_wcoss2_lua
 Changes to be committed:
	modified:   modulefiles/gefs/gefs_nemsio2nc.hera
	modified:   sorc/build_gefs_anom2_fcst.sh
	modified:   sorc/build_gefs_ensstat.sh
	modified:   sorc/build_gefs_nemsio2nc.sh
	modified:   sorc/build_gefs_nstgen.sh
	modified:   sorc/build_global_ensadd.sh
	modified:   sorc/build_global_ensppf.sh
	modified:   sorc/build_wave_stat.sh

Refs: #59

WalterKolczynski-NOAA commented on sorc/build_gefs_anom2_fcst.sh:
```diff
@@ -6,15 +6,20 @@ cwd=`pwd`
-source ../modulefiles/gefs/${progname}.$target
+if [[ $target == "wcoss2" ]]; then
+	module use ${cwd}/../modulefiles/gefs
+	module load ${progname}.${target}.lua
```
You don't need the `.lua` suffix here.

XianwuXue-NOAA replied:
Thanks for pointing this, I have  tested removing .lua, and it works well. I will put a comment for this.

WalterKolczynski-NOAA commented on sorc/build_gefs_anom2_fcst.sh:
```diff
@@ -6,15 +6,20 @@ cwd=`pwd`
 fi
 
-source ../modulefiles/gefs/${progname}.$target
+if [[ $target == "wcoss2" ]]; then
```
Why are you using two different methods for module loading?

XianwuXue-NOAA replied:
This is to make it compatible with other machines, such as Hera. After I change the modules to Lua on Hera, then it will be only one method for module loadings.

WalterKolczynski-NOAA replied:
I think you'd be better off just updating to lua for all machines at once.

XianwuXue-NOAA replied:
OK. I will do that.

XianwuXue-NOAA replied:
I have convert Hera module files to lua format and also unified the module loading methods

XianwuXue-NOAA committed:
Remove .lua suffix when loading modules

 On branch feature/dev_wcoss2_lua
 Changes to be committed:
	modified:   sorc/build_gefs_anom2_fcst.sh
	modified:   sorc/build_gefs_ensstat.sh
	modified:   sorc/build_gefs_nemsio2nc.sh
	modified:   sorc/build_gefs_nstgen.sh
	modified:   sorc/build_global_ensadd.sh
	modified:   sorc/build_global_ensppf.sh
	modified:   sorc/build_wave_stat.sh

Refs: #59

XianwuXue-NOAA committed:
Renmae modulefiles/gefs/*.hera to modulefiles/gefs/*.hera.lua

 On branch feature/dev_wcoss2_lua
 Changes to be committed:
	renamed:    modulefiles/gefs/gefs_anom2_fcst.hera -> modulefiles/gefs/gefs_anom2_fcst.hera.lua
	renamed:    modulefiles/gefs/gefs_ensstat.hera -> modulefiles/gefs/gefs_ensstat.hera.lua
	renamed:    modulefiles/gefs/gefs_nemsio2nc.hera -> modulefiles/gefs/gefs_nemsio2nc.hera.lua
	renamed:    modulefiles/gefs/gefs_nstgen.hera -> modulefiles/gefs/gefs_nstgen.hera.lua
	renamed:    modulefiles/gefs/global_ensadd.hera -> modulefiles/gefs/global_ensadd.hera.lua
	renamed:    modulefiles/gefs/global_ensppf.hera -> modulefiles/gefs/global_ensppf.hera.lua
	renamed:    modulefiles/gefs/wave_stat.hera -> modulefiles/gefs/wave_stat.hera.lua

Refs: #59

XianwuXue-NOAA committed:
Convert modulefiles on Hera to Lua format

 On branch feature/dev_wcoss2_lua
	modified:   modulefiles/gefs/gefs_anom2_fcst.hera.lua
	modified:   modulefiles/gefs/gefs_ensstat.hera.lua
	modified:   modulefiles/gefs/gefs_nemsio2nc.hera.lua
	modified:   modulefiles/gefs/gefs_nstgen.hera.lua
	modified:   modulefiles/gefs/global_ensadd.hera.lua
	modified:   modulefiles/gefs/global_ensppf.hera.lua
	modified:   modulefiles/gefs/wave_stat.hera.lua
	new file:   rocoto/dev/versions/build_hera.ver
	new file:   rocoto/dev/versions/run_hera.ver
	modified:   sorc/build_gefs_anom2_fcst.sh
	modified:   sorc/build_gefs_ensstat.sh
	modified:   sorc/build_gefs_nemsio2nc.sh
	modified:   sorc/build_gefs_nstgen.sh
	modified:   sorc/build_global-workflow.sh
	modified:   sorc/build_global_ensadd.sh
	modified:   sorc/build_global_ensppf.sh
	modified:   sorc/build_wave_stat.sh

Refs: #59

XianwuXue-NOAA committed:
Move modules for jet to rocoto/dev for future ref

 On branch feature/dev_wcoss2_lua
 Changes to be committed:
	renamed:    modulefiles/gefs/gefs_anom2_fcst.jet -> rocoto/dev/modulefiles/gefs/gefs_anom2_fcst.jet
	renamed:    modulefiles/gefs/gefs_ensstat.jet -> rocoto/dev/modulefiles/gefs/gefs_ensstat.jet
	renamed:    modulefiles/gefs/gefs_nstgen.jet -> rocoto/dev/modulefiles/gefs/gefs_nstgen.jet
	renamed:    modulefiles/gefs/global_ensadd.jet -> rocoto/dev/modulefiles/gefs/global_ensadd.jet
	renamed:    modulefiles/gefs/global_ensppf.jet -> rocoto/dev/modulefiles/gefs/global_ensppf.jet

Refs: #59

XianwuXue-NOAA committed:
Delte unused module on Orion

 On branch feature/dev_wcoss2_lua
	deleted:    modulefiles/gefs/gefs_nemsio2nc.orion

Refs: #59

XianwuXue-NOAA commented:
@WalterKolczynski-NOAA I have converted the module files for Hera and also unified the build_*.sh scripts

XianwuXue-NOAA commented:
@WalterKolczynski-NOAA When you have time, could you please review my changes? After you approved it, I will merge it and do the bugzilla issues. Thanks,
